docs/PowerAuth-Push-Server-1.4.0.md

```diff
@@ -0,0 +1,64 @@
+# Migration from 1.3.x to 1.4.x
+
+## New Message Inbox APIs
+
+You can review and consider implementing our [Message Inbox API](./Push-Server-API.md). Message inbox is a new feature
+of the push server that can be used to send targeted messages to the users to be read later.
+
+## Database Changes
+
+### Message Inbox Tables
+
+Message inbox requires a simple database structure below:
+
+#### PostgreSQL
+
+```sql
+CREATE TABLE push_inbox (
+    id VARCHAR(37) CONSTRAINT push_inbox_pk PRIMARY KEY,
+    user_id VARCHAR(255) NOT NULL,
+    subject TEXT NOT NULL,
+    body TEXT NOT NULL,
+    read BOOLEAN DEFAULT false,
+    timestamp_created TIMESTAMP NOT NULL,
+    timestamp_read TIMESTAMP
+);
+
+CREATE INDEX push_inbox_user ON push_inbox (user_id);
+CREATE INDEX push_inbox_user_read ON push_inbox (user_id, timestamp_created, read);
```

Comment:
I am missing following index:
- `user_id`, `read`

I think the index on (`user_id`, `timestamp_created`, `read`) is not very beneficial, because `timestamp_created` is not used after WHERE, it is only used in ORDER BY. The DB can sort results which are not too big (up to tens of thousands) fast without index. 

We certainly need a dedicated index for the select with `user_id` and `read` in WHERE condition.

Reply:
Good point, thank you! 👍